core-restore/src/test/scala/io/aiven/guardian/kafka/restore/RestoreClientInterfaceTest.scala

```diff
@@ -0,0 +1,150 @@
+    forAll(
+      kafkaDataWithTimePeriodsAndPickedRecordGen(padTimestampsMillis =
+                                                   Range.inclusive(1, ChronoUnit.HOURS.getDuration.toMillis.toInt),
+                                                 min = 1000,
+                                                 max = 1000
+      )
+    ) { (kafkaDataWithTimePeriodAndPickedRecord: KafkaDataWithTimePeriodAndPickedRecord) =>
+      implicit val mockedKafkaProducerInterface: MockedKafkaProducerInterface = new MockedKafkaProducerInterface
+      implicit val kafkaClusterConfig: KafkaClusterConfig =
+        KafkaClusterConfig(kafkaDataWithTimePeriodAndPickedRecord.topics)
+      val pickedTime = kafkaDataWithTimePeriodAndPickedRecord.picked.toOffsetDateTime
+      implicit val restoreConfig: RestoreConfig =
+        RestoreConfig(Some(pickedTime), None)
+
+      val backupMock =
+        new MockedBackupClientInterfaceWithMockedKafkaData(Source(kafkaDataWithTimePeriodAndPickedRecord.data),
+                                                           PeriodFromFirst(
+                                                             ChronoUnit.SECONDS.getDuration.toScala
+                                                           ),
+                                                           compression
+        )
+
+      backupMock.clear()
+      val calculatedFuture = for {
+        _ <- backupMock.backup.run()
+        _ <- akka.pattern.after(AkkaStreamInitializationConstant)(Future.successful(()))
+        processedRecords = backupMock.mergeBackedUpData(compression = compression.map(_.`type`))
+        restoreMock      = new MockedRestoreClientInterface(processedRecords.toMap)
+        keys <- restoreMock.finalKeys
+      } yield keys
+
+      val result = calculatedFuture.futureValue
+
+      Inspectors.forEvery(result.drop(1)) { key =>
+        (Utils.keyToOffsetDateTime(key) >= pickedTime) must be(true)
+      }
+
+    }
+  }
+
+  property("Round-trip with backup and restore") {
+    forAll(kafkaDataWithTimePeriodsGen()) { (kafkaDataWithTimePeriod: KafkaDataWithTimePeriod) =>
+      implicit val restoreConfig: RestoreConfig                               = RestoreConfig.empty
+      implicit val mockedKafkaProducerInterface: MockedKafkaProducerInterface = new MockedKafkaProducerInterface
+      implicit val kafkaClusterConfig: KafkaClusterConfig = KafkaClusterConfig(kafkaDataWithTimePeriod.topics)
+
+      val backupMock =
+        new MockedBackupClientInterfaceWithMockedKafkaData(Source(kafkaDataWithTimePeriod.data),
+                                                           PeriodFromFirst(kafkaDataWithTimePeriod.periodSlice),
+                                                           compression
+        )
+
+      backupMock.clear()
+      val calculatedFuture = for {
+        _ <- backupMock.backup.run()
+        _ <- akka.pattern.after(10 seconds)(Future.successful(()))
+        processedRecords = backupMock.mergeBackedUpData()
```

Comment:
We are not passing in the `compression` here because its the job of the actual `RestoreClient` to decompress the backed up data later on.